panels/config/zwave/zwave-log.html

```diff
@@ -16,16 +16,34 @@
         margin: 0 auto;
         max-width: 600px;
       }
+
+      .device-picker {
+        padding-left: 24px;
+        padding-right: 24px;
+        padding-bottom: 24px;
+      }
     </style>
     <ha-config-section is-wide='[[isWide]]'>
       <span slot='header'>OZW Log</span>
       <paper-card>
-        <div class='help-text'>
-          <pre>[[ozwLogs]]</pre>
-        </div>
-        <div class="card-actions">
-          <paper-button raised on-tap='refreshLog'>Refresh</paper-button>
-        </div>
+      <div class='device-picker'>
+        <paper-input
+          label="Number of last log lines."
+          type=number
+          min=0
+          max=1000
+          step=10
+          value={{numLogLines}}>
+        </paper-input>
+      </div>
+      <div class="card-actions">
+        <paper-button raised on-tap='refreshLog'>Refresh</paper-button>
+      </div>
+      <div class='help-text'>
+        <template is='dom-repeat' items='[[ozwLogs]]' as='state'>
```

Comment:
If you're going to create a <pre> tag per line, why even break it up per line at all on the backend ? Just return a text and just set that to a single `<pre>` tag.